analyzers/tests/SonarAnalyzer.Test/TestCases/PropertyGetterWithThrow.cs

```diff
@@ -84,5 +84,16 @@ public int MyProperty8
             {
             }
         }
+        public int MyProperty9
+        {
+            get
+            {
+                throw FactoryMethod(); // Compliant
+            }
+        }
+        private NotSupportedException FactoryMethod()
+        {
+            return new NotSupportedException();
```

Comment:
Add one more testcase that looks exactly like this, but returns `new Exception()` and is expected to raise.

Reply:
Good idea. With the simplification you suggested, we are then arguably testing that the behaviour of semanticModel, but I see that as a good thing, as we are relying on it.